components/calendar/src/date.rs

```diff
@@ -362,3 +377,8 @@ impl<A: AsCalendar + Clone> Clone for Date<A> {
         }
     }
 }
+
+impl<A: AsCalendar + Copy> Copy for Date<A> where
+    <<A as AsCalendar>::Calendar as Calendar>::DateInner: Copy
+{
+}
```

Comment:
Nit: I'm not a fan of bounds in both the type signature and `where`

```suggestion
impl<A> Copy for Date<A>
where
    A: AsCalendar + Copy,
    <<A as AsCalendar>::Calendar as Calendar>::DateInner: Copy
{
}
```